### PR TITLE
Added Docker image generation to the core build system

### PR DESCRIPTION
**Pull request by sumerjabri**
### Ticket reference or the full description of what's in the PR
https://github.com/craftercms/craftercms/issues/6781

### CODE REVIEW

File: docker.gradle [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+		"apache-tomcat/webapps/crafter-search*",
+		"apache-tomcat/webapps/crafter-social*",
+		"apache-tomcat/webapps/crafter-profile*",
+		"apache-tomcat/webapps/ROOT",
+		"apache-tomcat/webapps/studio",
+		"**/*.pid"
+	]
+}
+
+task downloadLibSsl(type: Download) {
```
Comment (sumerjabri) [inline docker.gradle:87]: @phuongnq is this still needed?

File: docker.gradle [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+	src 'https://downloads.craftercms.org/misc/libssl.so.1.1'
+	overwrite false
+	dest "${libDir}/libssl.so.1.1"
+}
+
+task downloadLibCrypto(type: Download) {
```
Comment (sumerjabri) [inline docker.gradle:93]: @phuongnq is this still needed?

File: docker.gradle [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+
+ext {
+	imagesDir = "${projectDir}/resources/docker/images"
+	imagesBuildDir = "${buildDir}/images"
+	tempDir = "${buildDir}/temp"
+	libDir = "${tempDir}/lib"
+
+	authoringBuildTarget = 'main'
+
+	// TODO: Move this to EE
```
Comment (sumerjabri) [inline docker.gradle:39]: Move this to EE

File: docker.gradle (hunk 1 above)
Comment (phuongnq) [reply docker.gradle:82]: @sumerjabri We don't need that for the `develop` branch where MariaDB 11 is used. I have sent PR for this (also update to Gradle 8.7): https://github.com/craftercms/docker-images/pull/93

File: docker.gradle (hunk 2 above)
Comment (phuongnq) [reply docker.gradle:88]: @sumerjabri We don't need that for the `develop` branch where MariaDB 11 is used. I have sent PR for this (also update to Gradle 8.7): https://github.com/craftercms/docker-images/pull/93

File: resources/docker/images/authoring/tomcat/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+FROM eclipse-temurin:17-jdk-alpine AS main
+
+############# INSTALL NATIVE TOMCAT #############
+ENV CATALINA_HOME=/usr/local/tomcat \
+    TOMCAT_MAJOR="9" \
```
Comment (phuongnq) [inline resources/docker/images/authoring/tomcat/Dockerfile:18]: The Dockerfile seems incorrect for `develop` branch where we should use Java 21 and Tomcat 10. We have the latest on https://github.com/craftercms/docker-images/tree/develop

**Pull request merged**